lib/dialects/mssql/blocks.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+var _ = require('underscore');
+
+module.exports = function(dialect) {
+	var parentValueBlock = dialect.blocks.get('value');
+
+	dialect.blocks.set('value', function(params) {
+		var value = params.value;
+
+		var result;
+		if (_.isBoolean(value)) {
+			result = String(Number(value));
+		} else {
+			result = parentValueBlock(params);
+		}
+
+		return result;
+	});
+
+	dialect.blocks.set('limit', function(params) {
+		var result = '';
+		if (_.isUndefined(params.offset)) {
+			result = 'top(' + dialect.builder._pushValue(params.limit) + ')';
+		}
+		return result;
+	});
+
+	dialect.blocks.set('offset', function(params) {
+		var prefix = (!params.sort) ? 'order by 1 ' : '';
```

Comment:
https://github.com/2do2go/json-sql/pull/26#discussion_r116425720
https://github.com/2do2go/json-sql/pull/26#discussion_r116426693

This function was refactored to clean up the program flow. `offset` case fixed